src/PerformanceTests/Utils/BatchHelper.cs

```diff
@@ -19,6 +19,19 @@ public Task Batch(int count, Func<int, Task> action)
             return Task.WhenAll(sends);
         }
     }
+    public class TaskWhenAllTaskRun : IBatchHelper
+    {
+        public Task Batch(int count, Func<int, Task> action)
+        {
+            var sends = new Task[count];
+            for (var i = 0; i < count; i++)
+            {
+                var i2 = i;
+                sends[i] = Task.Run(() => action(i2));
```

Comment:
this isn't going to perform well for large `count`.

Reply:
Could you define large? Tested this with size of 150,000 and worked flawlessly, see https://github.com/Particular/NServiceBus.Transport.Msmq/pull/31

Reply:
I think the overhead generated by that many task.run's won't be that effective as more refined solutions (e.g. https://docs.particular.net/nservicebus/handlers/async-handlers#concurrency-large-amount-of-concurrent-message-operations) . But since this is non-performance relevant code, I guess that's good enough.